src/Serialization/main.jl

```diff
@@ -204,14 +208,22 @@ function load_type_dispatch(s::DeserializerState, ::Type{T}, dict::Dict) where T
 
     Base.issingletontype(T) && return T()
 
-    result = load_internal(s, T, dict[:data])
+    if parent != nothing
```

Comment:
Minor efficiency suggestion:
```suggestion
    if parent !== nothing
```